### PR TITLE
Feature/redis cache

### PR DESCRIPTION
**Pull request by ernestman28**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: infrastructure/main.tf [hunk 1]
```diff
@@ -110,3 +114,29 @@ data "azurerm_key_vault_secret" "oauth2_secret" {
...
+
+data "azurerm_key_vault_secret" "redis_secret" {
+  name      = "redis-secret"
+  key_vault_id = "${data.azurerm_key_vault.div_key_vault.id}"
+}
+
+resource "azurerm_key_vault_secret" "redis_connection_string" {
+  name = "${var.component}-redis-connection-string"
+  value = "redis://ignore:${urlencode(module.redis-cache.access_key)}@${module.redis-cache.host_name}:${module.redis-cache.redis_port}?tls=true"
+  key_vault_id = "${data.azurerm_key_vault.div_key_vault.id}"
```
Comment (adusumillipraveen) [inline infrastructure/main.tf]: some references to div, want to fix them ?

Comment (ernestman28) [reply infrastructure/main.tf]: cool, i will changed. thx for spotting it

Comment (ernestman28) [reply infrastructure/main.tf]: changed

File: infrastructure/main.tf [hunk 2]
```diff
@@ -88,6 +88,10 @@ module "app" {
 
         WEBSITE_NODE_DEFAULT_VERSION  = "12.13.0"
 
+        // Redis Cloud
+        REDISCLOUD_URL = "redis://ignore:${urlencode(module.redis-cache.access_key)}@${module.redis-cache.host_name}:${module.redis-cache.redis_port}?tls=true"
+        REDIS_ENCRYPTION_SECRET = "${data.azurerm_key_vault_secret.redis_secret.value}"
```
Comment (adusumillipraveen) [inline infrastructure/main.tf]:
```suggestion
        REDIS_ENCRYPTION_SECRET = "${module.redis-cache.access_key}"
```

Comment (adusumillipraveen) [reply infrastructure/main.tf]: It might fail first time if not

**Pull request merged**